nixos/modules/services/torrent/transmission.nix

```diff
@@ -6,26 +6,31 @@ let
   cfg = config.services.transmission;
   apparmor = config.security.apparmor.enable;
 
-  homeDir = "/var/lib/transmission";
-  downloadDir = "${homeDir}/Downloads";
-  incompleteDir = "${homeDir}/.incomplete";
-
-  settingsDir = "${homeDir}/.config/transmission-daemon";
-  settingsFile = pkgs.writeText "settings.json" (builtins.toJSON fullSettings);
-
-  # Strings must be quoted, ints and bools must not (for settings.json).
-  toOption = x:
-    if x == true then "true"
-    else if x == false then "false"
-    else if isInt x then toString x
-    else toString ''"${x}"'';
-
-  # for users in group "transmission" to have access to torrents
-  fullSettings = { umask = 2; download-dir = downloadDir; incomplete-dir = incompleteDir; } // cfg.settings;
+  fullSettings = {
+    "use-incomplete-dir" = cfg.useIncompleteDir;
+    "incomplete-dir"     = cfg.incompleteDir;
+    "rpc-whitelist"      = concatStringsSep "," cfg.rpcWhitelist;
+    "rpc-whitelist-enabled" = (builtins.length cfg.rpcWhitelist > 0);
+    "download-dir"       = cfg.downloadDir;
+  } // cfg.settings;
+
+  _baseDir = "/var/lib/transmission";
+  _settingsDir = "$_baseDir/.config/transmission-daemon";
```

Comment:
I wonder: what is the intention behind using the `$_baseDir` syntax? Won't it expand to the literal string `"$_baseDir/.config/transmission-daemon"`?